simplipy/api.py

```diff
@@ -77,7 +82,16 @@ def __init__(
         self.user_id: int | None = None
         self.websocket: WebsocketClient | None = None
 
-        self.async_request = self._wrap_request_method(self._request_retries)
+        self.async_request = self._wrap_request_method(
+            request_retries=self._request_retries,
+            giveup_func=self.is_fatal_error([401, 409]),
+            request_func=self._async_api_request,
+        )
+        self.async_media = self._wrap_request_method(
+            request_retries=self._media_retries,
+            giveup_func=self.is_fatal_error([401, 409, 404]),
```

Comment:
Instead of passing the function (which remains the same), how about just passing the error codes you care about?

Reply:
Done